config/cmake/modules/MfemCmakeUtilities.cmake

```diff
@@ -874,18 +874,22 @@ function(mfem_export_mk_files)
          MFEM_CXXFLAGS)
   set(MFEM_TPLFLAGS "")
   foreach(dir ${TPL_INCLUDE_DIRS})
-    set(MFEM_TPLFLAGS "${MFEM_TPLFLAGS} -I${dir}")
+    set(MFEM_TPLFLAGS "${MFEM_TPLFLAGS} -I${dir}" CACHE STRING "" FORCE)
   endforeach()
   # TODO: MFEM_TPLFLAGS: add other TPL flags, in addition to the -I flags.
-  set(MFEM_INCFLAGS "-I\$(MFEM_INC_DIR) \$(MFEM_TPLFLAGS)")
+  set(MFEM_INCFLAGS "-I\$(MFEM_INC_DIR) \$(MFEM_TPLFLAGS)" CACHE STRING "" FORCE)
   set(MFEM_PICFLAG "")
-  if (BUILD_SHARED_LIBS)
-    set(MFEM_PICFLAG "${CMAKE_SHARED_LIBRARY_CXX_FLAGS}")
+  if (BUILD_SHARED_LIBS OR MFEM_USE_JIT)
+    if (MFEM_USE_CUDA)
+       set(MFEM_PICFLAG "-Xcompiler=${CMAKE_SHARED_LIBRARY_CXX_FLAGS}")
+   else()
+       set(MFEM_PICFLAG "${CMAKE_SHARED_LIBRARY_CXX_FLAGS}")
+   endif(MFEM_USE_CUDA)
   endif()
   set(MFEM_FLAGS "\$(MFEM_CPPFLAGS) \$(MFEM_CXXFLAGS) \$(MFEM_INCFLAGS)")
   # TPL link flags: set below
   set(MFEM_EXT_LIBS "")
-  if (BUILD_SHARED_LIBS)
+  if (BUILD_SHARED_LIBS OR MFEM_USE_JIT)
```

Comment:
JIT implies shared, so there's no need to check for JIT.

Reply:
`done`